LIBZ-NG ZCONF.H breaks SYSTEMD Master

As of this afternoon, 'zconf.h' breaks the main branch build of 'systemd' i.e.

git clone git clone https://github.com/systemd/systemd.git
meson setup build -Dstatic-libsystemd=true -Dstatic-libudev=true

The trace is as follows:-

[1173/2731] Compiling C object src/import/libimport-common.a.p/qcow2-util.c.o
FAILED: src/import/libimport-common.a.p/qcow2-util.c.o
ccache cc -Isrc/import/libimport-common.a.p -Isrc/import -I../src/import -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -I/usr/local/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Warray-bounds -Warray-bounds=2 -Wdate-time -Wendif-labels -Werror=format=2 -Werror=format-signedness -Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=missing-declarations -Werror=missing-prototypes -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wunused-function -Wwrite-strings -Wzero-length-bounds -fdiagnostics-show-option -fno-common -fstack-protector -fstack-protector-strong --param=ssp-buffer-size=4 -Wno-unused-result -Werror=shadow -fPIC -fno-strict-aliasing -fvisibility=hidden -fno-omit-frame-pointer -include config.h -MD -MQ src/import/libimport-common.a.p/qcow2-util.c.o -MF src/import/libimport-common.a.p/qcow2-util.c.o.d -o src/import/libimport-common.a.p/qcow2-util.c.o -c ../src/import/qcow2-util.c
In file included from /usr/local/include/zlib.h:41,
                 from ../src/import/qcow2-util.c:3:
/usr/local/include/zconf.h:28:9: error: unknown type name ‘size_t’
   28 | typedef size_t z_size_t;
      |         ^~~~~~
[1182/2731] Generating symbol file src/shared/libsystemd-shared-253.so.p/libsystemd-shared-253.so.symbols
ninja: build stopped: subcommand failed.

Whereas the 'systemd' main branch still compiles successfully against the latest 'zconf.h' from 'https://github.com/madler/zlib' master.

Platform is Linux btw...

Apologies in advance if this is deemed to be the responsibility of the 'systemd' project but just thought you should be given the heads up in any case.

--- a/zconf.h.in
+++ b/zconf.h.in
@@ -25,8 +25,6 @@
 #  define z_const
 #endif
 
-typedef size_t z_size_t;
-
 /* Maximum value for memLevel in deflateInit2 */
 #ifndef MAX_MEM_LEVEL
 #  define MAX_MEM_LEVEL 9
@@ -199,5 +197,7 @@ typedef PTRDIFF_TYPE ptrdiff_t;
 #    define z_off64_t z_off_t
 #  endif
 #endif
+
+typedef size_t z_size_t;
 
 #endif /* ZCONF_H */

